src/element-style-observer.js

```diff
@@ -109,7 +109,29 @@ export default class ElementStyleObserver {
 	 * @param {TransitionEvent} event
 	 */
 	async handleEvent (event) {
-		if (!this.properties.has(event.propertyName)) {
+		if (event.type === "animationstart") {
+			let animation = document.getAnimations().find(a => a.effect.target === this.target);
+			let properties = this.getObservedAnimationProperties(animation);
```

Comment:
I’m not sure if this needs to be a separate function. It's only called once, and it seems to niche to be useful outside of this (but perhaps you plan to call it from somewhere else too?)

Reply:
Yes, you are right. We don't need a separate function for this. I just thought it would make the code a bit cleaner. But it looks OK even with the inlined code.